### PR TITLE
docs(api): Document update member's org/team role

### PR DESCRIPTION
**Pull request by schew2381**
## Duplicate Functionality
There is another endpoint called `OrganizationMemberTeamDetailsEndpoint` that can also update a single member's Team Role like the endpoint this PR documents. This is the API that's called when you're actually on the `Team` tab pictured below.
Screenshot 2023-08-04 at 2 12 11 PM

I decided to not document the duplicate endpoint because it runs into the same permission scopes issue as https://github.com/getsentry/sentry/pull/50900/ with regards to Team Admins. I also wanted to avoid 2 endpoints that do the same thing in our documentation. 

Unfortunately, this means that Team Admins that are regular org members cannot change roles for the teams they're on. I don't believe this is a huge issue because Team Admins do this through the UI, and changing team roles is not as common as adding/removing team members.

## Endpoint Title
This is titled `Update an Organization Member's Roles` instead of `Update an Organization Member` because the only other thing this endpoint can do is resend an invite to an invited member who hasn't joined yet. I don't believe this is necessary to document, and including `roles` in the source makes the endpoint much easier to find.

Screenshot 2023-08-08 at 1 56 45 PM

## Garbage in Response
Unfortunately we interact with roles on the frontend by piping them through API responses. This leads to extremely long responses where the roles by themselves make up ~80% of body. Ideally we would add role information to the frontend, but that is a very difficult problem. 

Because the frontend expects roles, the `TypedDict` return type for the PUT response serializer cannot have these roles as optional. The example response is forced to include them or else the example validator errors. Below is what the roles look like in the example (very little is shown).

Screenshot 2023-08-02 at 6 30 12 PM
Screenshot 2023-08-02 at 6 30 26 PM

### CODE REVIEW

File: src/sentry/apidocs/parameters.py [hunk 1]
```diff
@@ -234,7 +285,7 @@ class ProjectParams:
 and IPv6 (``::1``) addresses.
 - `filtered-transaction`: Filter out transactions for healthcheck and ping endpoints.
 - `web-crawlers`: Filter out known web crawlers. Some crawlers may execute pages in incompatible
-ways which then cause errors that are unlikely to be seen by a normal user.
+ways which cause errors that are unlikely to be seen by a normal user.
```
Comment (schew2381) [inline src/sentry/apidocs/parameters.py:288]: nit: shorten line to make docs look better

File: src/sentry/apidocs/parameters.py [hunk 2]
```diff
@@ -319,7 +370,7 @@ class ProjectParams:
         location="query",
         required=False,
         type=inline_serializer(
-            name="OptionsSerializer",
+            name="ProjectOptionsSerializer",
```
Comment (schew2381) [inline src/sentry/apidocs/parameters.py:373]: nit: Rename for clarity

File: src/sentry/api/endpoints/organization_member/requests/invite/details.py [hunk 3]
```diff
@@ -8,14 +8,14 @@
 from sentry.api.base import region_silo_endpoint
 from sentry.api.bases import OrganizationMemberEndpoint
 from sentry.api.bases.organization import OrganizationPermission
+from sentry.api.endpoints.organization_member.index import OrganizationMemberSerializer
```
Comment (schew2381) [inline src/sentry/api/endpoints/organization_member/requests/invite/details.py:11]: nit: switch to absolute import

File: src/sentry/api/endpoints/organization_member/requests/invite/index.py [hunk 4]
```diff
@@ -6,6 +6,7 @@
 from sentry import audit_log, roles
 from sentry.api.base import region_silo_endpoint
 from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.api.endpoints.organization_member.index import OrganizationMemberSerializer
```
Comment (schew2381) [inline src/sentry/api/endpoints/organization_member/requests/invite/index.py:9]: nit: switch to absolute import

**Review by dashed — APPROVED** [review]

**Pull request merged**